### ISSUE TITLE
syntect_server: Occasionally terminating with exit code 143 in production

### ISSUE THREAD

**Issue by keegancsmith** [sourcegraph/sourcegraph#3164]
I was using sourcegraph.com and syntax highlighting was not working. After a few refreshes it started to work. This is the error reported by kubernetes:

```
Last State:    Terminated
      Reason:      Error
      Message:     hread '<unnamed>' panicked at 'Can only call resolve on linked references: ByScope { scope: <source.javascript>, sub_context: None }', /home/rust/.cargo/git/checkouts/syntect-eae6a61cc0258c3f/c9fc533/src/parsing/syntax_definition.rs:205:18
thread '<unnamed>' panicked at 'Can only call resolve on linked references: ByScope { scope: <source.javascript>, sub_context: None }', /home/rust/.cargo/git/checkouts/syntect-eae6a61cc0258c3f/c9fc533/src/parsing/syntax_definition.rs:205:18
Error: Failed to write response: Os { code: 32, kind: BrokenPipe, message: "Broken pipe" }.
thread '<unnamed>' panicked at 'Can only call resolve on linked references: ByScope { scope: <source.scss>, sub_context: None }', /home/rust/.cargo/git/checkouts/syntect-eae6a61cc0258c3f/c9fc533/src/parsing/syntax_definition.rs:205:18
thread '<unnamed>' panicked at 'Can only call resolve on linked references: ByScope { scope: <source.javascript>, sub_context: None }', /home/rust/.cargo/git/checkouts/syntect-eae6a61cc0258c3f/c9fc533/src/parsing/syntax_definition.rs:205:18
thread '<unnamed>' panicked at 'Can only call resolve on linked references: ByScope { scope: <source.javascript>, sub_context: None }', /home/rust/.cargo/git/checkouts/syntect-eae6a61cc0258c3f/c9fc533/src/parsing/syntax_definition.rs:205:18
thread '<unnamed>' panicked at 'Can only call resolve on linked references: ByScope { scope: <source.javascript>, sub_context: None }', /home/rust/.cargo/git/checkouts/syntect-eae6a61cc0258c3f/c9fc533/src/parsing/syntax_definition.rs:205:18
thread '<unnamed>' panicked at 'Can only call resolve on linked references: ByScope { scope: <source.javascript>, sub_context: None }', /home/rust/.cargo/git/checkouts/syntect-eae6a61cc0258c3f/c9fc533/src/parsing/syntax_definition.rs:205:18
Error: Failed to write response: Os { code: 32, kind: BrokenPipe, message: "Broken pipe" }.
Error: Failed to write response: Os { code: 32, kind: BrokenPipe, message: "Broken pipe" }.
Error: Failed to write response: Os { code: 32, kind: BrokenPipe, message: "Broken pipe" }.

      Exit Code:    143
      Started:      Mon, 01 Apr 2019 12:10:41 +0200
      Finished:     Tue, 02 Apr 2019 10:19:21 +0200
```

Additionally given syntect_server is stateless it would probably benefit from 2 replicas + a liveness check.

**Comment by slimsag** [comment]
https://app.hubspot.com/contacts/2762526/company/407948923 ran into this as well, fix incoming

**Comment by slimsag** [comment]
Fixed and rolled out to prod

**Comment by slimsag** [comment]
& included in 3.3
